maven-surefire-plugin version 2.19.1 not supported

Using maven-surefire-plugin 2.18 works as expected, however using the latest version of 2.19.1 will result in the following stack trace during Maven clean install:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project html: Execution default-test of goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test failed: There was an error in the forked process
[ERROR] org.apache.maven.surefire.util.SurefireReflectionException: java.lang.NoSuchMethodError: org.apache.maven.surefire.providerapi.ProviderParameters.getProviderProperties()Ljava/util/Properties;
[ERROR] at org.apache.maven.surefire.util.ReflectionUtils.instantiateOneArg(ReflectionUtils.java:135)
[ERROR] at org.apache.maven.surefire.booter.ForkedBooter.createProviderInCurrentClassloader(ForkedBooter.java:321)
[ERROR] at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
[ERROR] at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
[ERROR] at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
[ERROR] Caused by: java.lang.NoSuchMethodError: org.apache.maven.surefire.providerapi.ProviderParameters.getProviderProperties()Ljava/util/Properties;
[ERROR] at org.forgerock.cuppa.maven.surefire.CuppaSurefireProvider.<init>(CuppaSurefireProvider.java:52)
[ERROR] at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[ERROR] at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[ERROR] at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[ERROR] at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
[ERROR] at org.apache.maven.surefire.util.ReflectionUtils.instantiateOneArg(ReflectionUtils.java:131)
[ERROR] ... 4 more
[ERROR] -> [Help 1]
```

**Workaround:** Switch back to 2.1

--- a/build.gradle
+++ b/build.gradle
@@ -1,4 +1,4 @@
-allprojects  {
+allprojects {
   apply plugin: 'maven'
 
   group = 'org.forgerock.cuppa'
@@ -18,8 +18,22 @@ subprojects {
     mavenCentral()
   }
 
+  configurations {
+    provided
+  }
+
+  sourceSets {
+    main {
+      compileClasspath += configurations.provided
+    }
+    test {
+      compileClasspath += configurations.provided
+    }
+  }
+
   javadoc {
     options.stylesheetFile = new File(rootDir, "docs/css/javadoc.css")
+    classpath += configurations.provided
   }
 
   dependencies {

--- a/cuppa-junit/build.gradle
+++ b/cuppa-junit/build.gradle
@@ -2,5 +2,5 @@ description = 'Cuppa JUnit Integration'
 
 dependencies {
     compile project(':cuppa')
-    compile group: 'junit', name: 'junit', version: '4.12'
+    provided group: 'junit', name: 'junit', version: '4.12', transitive: false
 }

--- a/cuppa-surefire/build.gradle
+++ b/cuppa-surefire/build.gradle
@@ -1,8 +1,7 @@
-
 description = 'Cuppa Maven Surefire Integration'
 
 dependencies {
     compile project(':cuppa')
-    compile group: 'org.apache.maven.surefire', name: 'surefire-api', version: '2.14.1'
-    compile group: 'org.apache.maven.surefire', name: 'common-java5', version: '2.14.1'
+    provided group: 'org.apache.maven.surefire', name: 'surefire-api', version: '2.19.1', transitive: false
+    provided group: 'org.apache.maven.surefire', name: 'common-java5', version: '2.19.1', transitive: false
 }

--- a/cuppa-surefire/src/main/java/org/forgerock/cuppa/maven/surefire/CuppaSurefireProvider.java
+++ b/cuppa-surefire/src/main/java/org/forgerock/cuppa/maven/surefire/CuppaSurefireProvider.java
@@ -18,9 +18,7 @@ package org.forgerock.cuppa.maven.surefire;
 
 import java.util.Arrays;
 import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Properties;
+import java.util.Map;
 import java.util.Set;
 import java.util.stream.Collectors;
 
@@ -49,16 +47,16 @@ public final class CuppaSurefireProvider extends AbstractProvider {
      */
     public CuppaSurefireProvider(ProviderParameters parameters) {
         this.providerParameters = parameters;
-        Properties properties = parameters.getProviderProperties();
+        Map<String, String> properties = parameters.getProviderProperties();
         tags = new Tags(getTags(properties), getExcludedTags(properties));
     }
 
-    private Set<String> getTags(Properties properties) {
-        String groups = properties.getProperty("groups");
+    private Set<String> getTags(Map<String, String> properties) {
+        String groups = properties.get("groups");
         if (groups == null) {
             groups = System.getProperty("groups");
         }
-        String tags = properties.getProperty("tags");
+        String tags = properties.get("tags");
         String overrideTags = System.getProperty("tags");
         return getTags(groups, overrideTags == null ? tags : overrideTags);
     }
@@ -76,12 +74,12 @@ public final class CuppaSurefireProvider extends AbstractProvider {
         }
     }
 
-    private Set<String> getExcludedTags(Properties properties) {
-        String excludedGroups = properties.getProperty("excludedGroups");
+    private Set<String> getExcludedTags(Map<String, String> properties) {
+        String excludedGroups = properties.get("excludedGroups");
         if (excludedGroups == null) {
             excludedGroups = System.getProperty("excludedGroups");
         }
-        String excludedTags = properties.getProperty("excludedTags");
+        String excludedTags = properties.get("excludedTags");
         String overrideExcludedTags = System.getProperty("excludedTags");
         return getTags(excludedGroups, overrideExcludedTags == null ? excludedTags : overrideExcludedTags);
     }
@@ -95,17 +93,13 @@ public final class CuppaSurefireProvider extends AbstractProvider {
         ReporterFactory reporterFactory = providerParameters.getReporterFactory();
         RunListener listener = reporterFactory.createReporter();
         Runner runner = new Runner(tags);
-        TestBlock rootBlock = runner.defineTests(scanClasspathForTests());
+        TestBlock rootBlock = runner.defineTests(getSuites());
         runner.run(rootBlock, new CuppaSurefireReporter(listener));
         return reporterFactory.close();
     }
 
     @Override
-    public Iterator getSuites() {
-        return scanClasspathForTests().iterator();
-    }
-
-    private List<Class<?>> scanClasspathForTests() {
+    public Iterable<Class<?>> getSuites() {
         return Arrays.asList(providerParameters.getScanResult()
                 .applyFilter(clazz -> Arrays.stream(clazz.getAnnotations())
                                 .anyMatch(annotation -> Test.class.equals(annotation.annotationType())),

